File: tests/authoring/Blocks/Storybook.fs
// Licensed to Elasticsearch B.V under one or more agreements.
// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
// See the LICENSE file in the project root for more information
module ``AuthoringTests``.``block elements``.``storybook elements``

open Xunit
open authoring

type ``storybook missing reference`` () =
    static let markdown = Setup.Markdown """
:::{storybook}
:::
"""

    [<Fact>]
    let ``has error`` () = markdown |> hasError "requires :id: or :project:"

type ``storybook missing registry`` () =
    static let markdown = Setup.Markdown """
:::{storybook}
:id: kibana:shared_ux:components-button--regular
:::
"""

    [<Fact>]
    let ``has error`` () = markdown |> hasError "requires docset.yml storybook.registry"

File: tests/Elastic.Authoring.Tests/Blocks/Storybook.cs
// Licensed to Elasticsearch B.V under one or more agreements.
// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
// See the LICENSE file in the project root for more information

namespace Elastic.Authoring.Tests.Blocks.Storybook;

public class StorybookMissingReference : MarkdownTest
{
	protected override string Markdown => """
		:::{storybook}
		:::
		""";

	[Fact(DisplayName = "has error")]
	public async Task HasError() => await Docs.HasError("requires :id: or :project:");
}

public class StorybookMissingRegistry : MarkdownTest
{
	protected override string Markdown =>
		"""
		:::{storybook}
		:id: kibana:shared_ux:components-button--regular
		:::
		""";

	[Fact(DisplayName = "has error")]
	public async Task HasError() => await Docs.HasError("requires docset.yml storybook.registry");
}
